doc/sphinxext/docscrape.py

```diff
@@ -428,7 +429,11 @@ def __init__(self, func, role='func'):
             func, func_name = self.get_func()
             try:
                 # try to read signature
-                argspec = inspect.getargspec(func)
+                if 3 > sys.version_info[0]:
+                    argspec = inspect.getargspec(func)
+                else:
+                    # getargspec deprecated in python3
+                    argspec = inspect.getargspec(func)
```

Comment:
Yes, that's strange. We need to investigate this more.
Another point: it seems that the `warning.simplefilter("ignore", DeprecationWarning)` line is present is other places in this file, like [here](https://github.com/joblib/joblib/pull/375/files/f0d59bc491a814d91ed8987af837021474a77283#diff-8225e3cc054f5407acaad92d20bb0e79L257)
Can you remove them as well ?